src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -108,6 +108,10 @@ GcsServer::GcsServer(const ray::gcs::GcsServerConfig &config,
                   });
             });
       }),
+      event_aggregator_client_(std::make_unique<rpc::EventAggregatorClientImpl>(
+          config_.metrics_agent_port, client_call_manager_)),
+      ray_job_event_recorder_(std::make_unique<telemetry::RayJobEventRecorder>(
+          *event_aggregator_client_, io_context_provider_.GetDefaultIOContext())),
```

Comment:
what's the difference between an event "recorder" and an event "aggregator"

it seems like the "recorder" is just a client?